show/reboot_cause.py

```diff
@@ -23,6 +24,100 @@ def read_reboot_cause_file():
 
     return reboot_cause_dict
 
+# Function to fetch reboot cause data from database
+def fetch_reboot_cause_from_db(module_name):
+    table = []
+    r = []
+    wrapper = textwrap.TextWrapper(width=30)
+
+    # Read the previous reboot cause
+    if module_name == "all" or module_name == "SWITCH":
+        reboot_cause_dict = read_reboot_cause_file()
+        reboot_cause = reboot_cause_dict.get("cause", "Unknown")
+        reboot_user = reboot_cause_dict.get("user", "N/A")
+        reboot_time = reboot_cause_dict.get("time", "N/A")
+
+        r.append("SWITCH")
+        r.append(reboot_cause if reboot_cause else "")
+        r.append(reboot_time if reboot_time else "")
+        r.append(reboot_user if reboot_user else "")
+        table.append(r)
+
+        if module_name == "SWITCH":
+            return table
+
+    REBOOT_CAUSE_TABLE_NAME = "REBOOT_CAUSE"
+    TABLE_NAME_SEPARATOR = '|'
+    db = SonicV2Connector(host='127.0.0.1')
+    db.connect(db.STATE_DB, False)   # Make one attempt only
+    prefix = REBOOT_CAUSE_TABLE_NAME + TABLE_NAME_SEPARATOR
+    _hash = '{}{}'.format(prefix, '*')
+    table_keys = db.keys(db.STATE_DB, _hash)
+    if table_keys is not None:
+        table_keys.sort(reverse=True)
+
+    d = []
+    append = False
+    for tk in table_keys:
```

Comment:
The code starting from this line till the end of the function is identical to the code on lines 100-119. Please put it into the function to prevent duplication.

Reply:
Fixed